lib/ansible/modules/git.py

```diff
@@ -438,6 +447,17 @@ def get_version(module, git_path, dest, ref="HEAD"):
     return sha
 
 
+def ssh_supports_acceptnewhostkey(module):
+    ssh_path = module.get_bin_path('ssh', True)
+    supports_acceptnewhostkey = True
+    try:
+        cmd = [ssh_path, '-o', 'StrictHostKeyChecking=accept-new', '-V']
+        rc, stdout, stderr = module.run_command(cmd)
+    except Exception:
+        supports_acceptnewhostkey = False
```

Comment:
In general, try to avoid catching `Exception`. This can cause errors to be swallowed and make things harder to debug.

It's not necessaryto catch exceptions from this method. Most exceptions are handled internally by the method and result in module failure. Checking the value of `rc` is sufficient.

```suggestion
    cmd = [ssh_path, '-o', 'StrictHostKeyChecking=accept-new', '-V']
    rc, stdout, stderr = module.run_command(cmd)
    if rc != 0:
        supports_acceptnewhostkey = False
```

Reply:
Fixed in https://github.com/ansible/ansible/pull/70499/commits/b8a46435ed7faef1c8998e2d2e8e5c0d8ceca505.